cylc/flow/task_pool.py

```diff
@@ -266,12 +266,22 @@ def db_add_new_flow_rows(self, itask: TaskProxy) -> None:
         self.workflow_db_mgr.put_insert_task_outputs(itask)
 
     def add_to_pool(self, itask) -> None:
-        """Add a task to the pool."""
+        """Add a task to the pool, if not already added."""
 
         self.active_tasks.setdefault(itask.point, {})
+        if itask.identity in self.active_tasks[itask.point]:
+            # This is normal on restart with waiting parentless tasks that
+            # auto-spawned to the runahead limit before shutdown (e.g. if
+            # you shut down immediately after starting up paused). Each
+            # will be resurrected from the DB again *and* try to auto-spawn
+            # again because we don't record that parentless tasks spawned.
+            # (But debug-log it anyway for any unexpected cases).
+            LOG.debug(f"{itask.identity} not added to n=0: already exists")
+            return None
+
         self.active_tasks[itask.point][itask.identity] = itask
         self.active_tasks_changed = True
-        LOG.debug(f"[{itask}] added to the n=0 window")
+        LOG.info(f"[{itask}] added to the n=0 window")
```

Comment:
Deliberate change - it's important for users to know when tasks enter the n=0 window (otherwise it only gets logged as INFO at the first state change).